src/Core/Monitor/DefaultPrincipalResolver.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Security.Claims;
+using System.Security.Principal;
+
+namespace Gibraltar.Monitor
+{
+    /// <summary>
+    /// A simple default principal resolver
+    /// </summary>
+    public class DefaultPrincipalResolver : IPrincipalResolver
+    {
+        private readonly bool IsWindows;
+
+        public DefaultPrincipalResolver()
+        {
+            IsWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        }
+
+        /// <inheritdoc />
+        public IPrincipal ResolveCurrentPrincipal()
```

Comment:
I'd probably implement this as a `public bool TryResolveCurrentPrincipal(out IPrincipal principal)` if not having one is OK, in the interest of being explicit about `null`s.

Or have an `UnknownPrincipal` type with a single value that can be used wherever an actual `IPrincipal` can't be resolved.

Reply:
Good idea, that also makes it clear that no exception will be thrown if it can't be resolved.

Reply:
Done.